### DISCUSSION TITLE
Sqlanywhere custom-lib driver error java Class not found

### DISCUSSION THREAD

**Discussion by nannicini** [nextgenhealthcare/connect#5001] [Q&A]
Hi,
i would like to connect a database reader channel to Sqlanywhere 16 to another host
I try to use the sql sybase driver of mirthconnect but it does not work. 
So i try to use jconn4.jar e sajdbc4.jar 

I put this file into custom-lib and i see into settings->Resources the loaded libraries.

When i deploy the new channel mirthconnect return this error:

Failed to deploy channel Sybase
Caused by: java.lang.NoClassDefFoundError: java/sql/Driver
Caused by: java.lang.ClassNotFoundException: java.sql.Driver

Have you any idea to solve the problem?
The problem is in classpath?

**Comment by jonbartels** [comment]
What version of MC are you using?
Show your database connector settings (mask hostnames and passwords).
Show your JS code for DB connections if you are using JS instead of the GUI.

The 3.12 user guide on page 313 in the section "Editing Database Drivers" explains how to tell MC that new DB drivers exist.

https://www.nextgen.com/-/media/files/nextgen-connect/nextgen-connect-312-user-guide.pdf

**Reply by nannicini** [reply]
I use 3.12 version of MIrth Connect.
The same configuration (Custom driver, jconn4.jar and sajdbc4.jar into custom-lib directory) on Windows Server 2019 work perfectly. 
Using ubuntu 20.04 i have the problem of driver class.
<img width="301" alt="custom" src="https://user-images.githubusercontent.com/14890690/152580933-8354bb32-3309-4d32-9d9a-2d8ac9de1b95.png">

Caused by: java.lang.ClassNotFoundException: java.sql.Driver

I try to add the driver into table too
<img width="466" alt="driver" src="https://user-images.githubusercontent.com/14890690/152581082-48ef2409-f05d-4896-8c7c-5f477a9a90de.png">
but i have the same error.

I think that i have to use only the windows server....:(
Any suggestions?
Probably the openjdk dont work like java 1.8 by oracle?

**Reply by jonbartels** [reply]
It is good that it works in Windows. This means you have found the correct configs and file paths to use.

Check simple problems first - 
1. Does the Resource listing between Windows and Ubuntu match?
2. Does your channel on Ubuntu use the correct Resource directory as a dependency?
3. Are the file permissions on Ubuntu set so that the system user that runs the MC process can read the JAR files?
4. What specific Java versions are you using on Ubunutu and Windows?

**Reply by nannicini** [reply]
Hi thank you for your answer.

I check the 4 points:
1. The resources match beetween windows and ubuntu
2. The directory for dependency is correct in the two systems  /custom-lib. The resources tabs show the thew library loaded
3. On ubuntu the permission to access file into /custom-lib is 777
4. On ubuntu java version is openjdk 11.0.13 on Windows Java version is 1.8 (Oracle)

I think that the problem would be in java version or in the custom JAR file.
Thanks a lot

Andrea
